CMakeLists.txt

```diff
@@ -279,6 +279,18 @@ else()
   )
 endif ()
 
+find_program(RUN_CLANG_TIDY run-clang-tidy DOC "The path to run-clang-tidy")
+
+if(RUN_CLANG_TIDY)
```

Comment:
Is it a good idea to force running this tool if it is found, maybe we should do it manually instead?

Reply:
I intentionally did not force executing this and just added a command so that users can run this manually. It can generate false positives and we need to investigate some more and decide what checks to enable before we enforce anything.